### ISSUE TITLE
unknown userscript property (as complained by Firefox 73.0.1 when editing)

### ISSUE THREAD

**Issue by Wolf-at-SO** [jakub-g/greasemonkey-userscripts#7]
When I added the language codes, I saw a warning sign before line 9:

https://github.com/jakub-g/greasemonkey-userscripts/blob/1441976623c6316869ad882105efbd7a5ff2d0ec/wiki/langSwitcher.js#L9

To me, it seems that this line could be deleted to make FF happy. If this is intended at all ;-)

BTW: thanks so much for this helpful script :-)

**Comment by jakub-g** [comment]
Hi @Wolf-at-SO - fixed in https://github.com/jakub-g/greasemonkey-userscripts/commit/e92fa58eadfb9663579ba12f0b3f9a7673fad716#diff-05b18c1ec5336ceecde38fbb0f5896f3
Been a while since edited this script (almost 7 years). I almost forgot it exists :)

**Comment by Wolf-at-SO** [comment]
Great to see this fixed that fast!

**Comment by Wolf-at-SO** [comment]
... sorry for bothering: is this official source (for updates) still valid? https://userscripts-mirror.org/scripts/show/83393

**Comment by jakub-g** [comment]
Historically the code lived in `userscripts.org/scripts/show/83393` but several years ago the site became abandoned. Someone spidered the website and created `userscripts-mirror.org`, but it's just a static mirror from several years ago, not updated since.

I didn't upload the script anywhere else since then, so the official source is this github repository.

Having said that, when you modify the code yourself in Greasemonkey/Tampermonkey, it won't auto-update anymore (to avoid erasing your custom changes).
The way the script is customizable (via editing the source) is not the best one, because it kills the auto-updater. It should be done via a different way (`GM.setValue/GM.getValue` + UI to set up the script) but back in the time I wrote the script, I didn't know about it/it didn't exist.

Basically for the time being, your best option is to edit the script yourself.

One day I hopefully take time to extract the code as a standalone Firefox/Chrome extension and make the proper config UI instead of relying on code editing :)

**Comment by Wolf-at-SO** [comment]
Thanks for the long explanation. Do you know a way to get up-to-date with the current source *and* have auto-update enabled (like remove/import)? (I'm asking because I observed some flaws in the script - with redirected Wikipedia pages, I guess - that aren't reproducible for me any more.)

**Comment by jakub-g** [comment]
> (I'm asking because I observed some flaws in the script - with redirected Wikipedia pages, I guess - that aren't reproducible for me any more, and so I expect another update soon.)

Didn't understand what you mean

> Do you know a way to get up-to-date with the current source and have auto-update enabled

To get the newest version, you can uninstall the script and then install it again.

You can do it by navigating directly to the URL of the raw file on GitHub. Because filename ends with `.user.js`, if you have TamperMonkey installed, it will catch the request, and instead of opening the file in browser, it will show the installation prompt

Here's the URL to install:
https://raw.githubusercontent.com/jakub-g/greasemonkey-userscripts/master/wiki/langSwitcher.user.js

But after you install it and do manual modifications, the auto-update will stop working (anyway I don't plan to do any updates for now, so it's not a big deal ;) 

Feel free to submit patches if you find some problems with the code.

**Comment by Wolf-at-SO** [comment]
The install URL works, even without TamperMonkey, but as I see now, I *have to modify* it (sometimes it takes me a bit longer) for changing the languages to match my preferred set ;)

I'll now try to figure out how to do the [GM.setValue/GM.getValue + UI](https://wiki.greasespot.net/GM.getValue) stuff or watch out for the glitch I observed once.
